samples/bpf/protect_gpt_kern.c

```diff
@@ -0,0 +1,20 @@
+#include <linux/bpf.h>
+#include <linux/bio.h>
+#include <linux/blk_types.h>
+#include "bpf/bpf_helpers.h"
+#include "bpf_legacy.h"
+
+char _license[] SEC("license") = "GPL";
+
+#define GPT_SECTORS 34
+
+SEC("gpt_io_filter")
+int protect_gpt(struct bpf_io_request *io_req)
+{
+	if (io_req->sector_start < GPT_SECTORS && op_is_write(io_req->opf & REQ_OP_MASK))
```

Comment:
Nice program! But there are a lot of mutating ops other than writes (flush, discard, secure erase). I wonder if this should be opf != 0

Reply:
That makes sense, updated